internal/tiltfile/starkit/load_interceptor.go

```diff
@@ -11,4 +11,7 @@ type LoadInterceptor interface {
 	// Ensure the content is present in the path returned
 	// Returns "" if this interceptor doesn't act on this path
 	LocalPath(t *starlark.Thread, path string) (string, error)
+
+	// Optionally, a LoadInterceptor can record some info to send to analytics later
+	AnalyticsInfo() []AnalyticsInfo
```

Comment:
ya, this feels like a lot of layers of indirection

have you considered using the existing data model pattern for collecting this data?
https://github.com/tilt-dev/tilt/blob/master/internal/tiltfile/starkit/model.go

we have pretty good conventions for adding things to that data model and pulling it off at the end of execution in ways that prevent common mistakes.

Reply:
i honestly completely forgot about that part of tiltfile world. is https://github.com/tilt-dev/tilt/pull/4229 more like it? (def feels much better to me)

Reply:
:+1: